src/tags/object/Image/ImageSelection.js

```diff
@@ -0,0 +1,88 @@
+import { getParent, types } from 'mobx-state-tree';
+import { ImageSelectionPoint } from './ImageSelectionPoint';
+
+export const ImageSelection = types.model({
+  start: types.maybeNull(ImageSelectionPoint),
+  end: types.maybeNull(ImageSelectionPoint),
+}).views(self => {
+  return {
+    get obj() {
+      return getParent(self);
+    },
+    get annotation() {
+      return self.obj.annotation;
+    },
+    get highlightedNodeExists() {
+      return !!self.annotation.highlightedNode;
+    },
+    get isActive() {
+      return self.start && self.end;
+    },
+    get x() {
+      return Math.min((self.start.x * self.scale), (self.end.x * self.scale));
+    },
+    get y() {
+      return Math.min((self.start.y * self.scale), (self.end.y * self.scale));
+    },
+    get width() {
+      return Math.abs((self.end.x * self.scale) - (self.start.x * self.scale));
+    },
+    get height() {
+      return Math.abs((self.end.y * self.scale) - (self.start.y * self.scale));
+    },
+    get scale() {
+      return self.obj.zoomScale;
+    },
+    get bbox() {
+      const { start, end } = self;
+
+      return self.isActive ? {
+        left: Math.min(start.x, end.x),
+        top: Math.min(start.y, end.y),
+        right: Math.max(start.x, end.x),
+        bottom: Math.max(start.y, end.y),
+      } : null;
+    },
+    includesBbox(bbox) {
+      return self.isActive && bbox && self.bbox.left <= bbox.left && self.bbox.top <= bbox.top && self.bbox.right >= bbox.right && self.bbox.bottom >= bbox.bottom;
```

Comment:
minor tweak - to make it a little more readable
```suggestion
      if (!self.isActive || !bbox) return false;
      const isLeftOf = self.bbox.left <= bbox.left;
      const isAbove = self.bbox.top <= bbox.top;
      const isRightOf = self.bbox.right >= bbox.right;
      const isBelow = self.bbox.bottom >= bbox.bottom;
      
      return isLeftOf && isAbove && isRightOf && isBelow;
```

Reply:
nice! i'm not the author of this code, just split a giant `Image.js` into smaller pieces. gonna apply this one anyways. thanks